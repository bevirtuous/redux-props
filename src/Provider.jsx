import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StoreContext } from './context';

/**
 * The store provider.
 */
export class Provider extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    store: PropTypes.shape().isRequired,
  };

  /**
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);
    props.store.subscribe(() => this.forceUpdate());
  }

  /**
   * @return {JSX}
   */
  render() {
    const { store, children } = this.props;
    const value = {
      dispatch: store.dispatch,
      state: store.getState(),
    };

    return (
      <StoreContext.Provider value={value}>
        {children}
      </StoreContext.Provider>
    );
  }
}
