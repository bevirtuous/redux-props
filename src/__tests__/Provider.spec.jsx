import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from '../Provider';

const store = {
  subscribe: jest.fn(),
  dispatch() {},
  getState: jest.fn(),
};

describe('<Provider />', () => {
  it('should fail if no store is provided', (done) => {
    try {
      shallow(<Provider />);
      done('Did not error');
    } catch (e) {
      done();
    }
  });

  it('it should render with a proper store', () => {
    const wrapper = shallow((
      <Provider store={store}>
        <div>
          Some
        </div>
      </Provider>
    ));
    expect(store.subscribe).toHaveBeenCalledTimes(1);
    expect(store.getState).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
