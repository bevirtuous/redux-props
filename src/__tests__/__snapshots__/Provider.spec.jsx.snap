// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Provider /> it should render with a proper store 1`] = `
<[object Object]
  value={
    Object {
      "dispatch": [Function],
      "state": undefined,
    }
  }
>
  <div>
    Some
  </div>
</[object Object]>
`;
